### ISSUE TITLE
SublimeLinter 4.0.1 does not trigger eslint

### ISSUE THREAD

**Issue by josezulu** [SublimeLinter/SublimeLinter-eslint#203]
So SublimeLinter updated to the new version... after fixing my user settings: csslinter works, htmlhint works, but eslint doesn't seem to do anything, not even erroring.

eslint worked just fine a few hours ago before this update to 4.0.1.

I have SublimeLinter-eslint 4.0.0 installed, I see in the console when ST3 loads up:
`reloading plugin SublimeLinter-eslint.linter`

This is on the sublime linter settings user file, inside "linters: { }":
`"eslint": {`
`"disable": false,`
`       "args": [],`
`       "excludes": []`
`       }`

NPM package installed: eslint@4.18.1

I have debug set to true.

When I open any javascript file (.js), sublimelinter-eslint doesn't trigger anything, I can't see anything happening on the console, not even errors (lint mode is on "load_save"), nothing happens either when I lint the file manually.

**Comment by braver** [comment]
What output do you see in the console (with debug) when you run "lint this view" on a javascript file?

**Comment by josezulu** [comment]
Absolutely nothing. There is no output.

When run "lint this view" on an html file, I see stuff going on, ending in:
`SublimeLinter: #2 SublimeLinter.sublime-package:809: htmlhint output:
 []`

**Comment by braver** [comment]
> I see stuff going on

Er... what stuff?

**Comment by josezulu** [comment]
Sorry, here is a summary what I see when manually linting an html file:
`SublimeLinter: SublimeLinter.sublime-package:252: detected syntax: html`
`SublimeLinter: #1 SublimeLinter.sublime-package:782: 'htmlhint' is linting 'index.html'`
`...`
`SublimeLinter: #2 SublimeLinter.sublime-package:809: htmlhint output: []`

When I do that on a .js file, nothing really happens in the console.

**Comment by kaste** [comment]
We already know or have reports that on load is broken, it is only on save currently. 🙄 Please, change the mode to 'background'. Restart Sublime with debug still true and post the log. 👍 ?

**Comment by josezulu** [comment]
@kaste I'm sorry if I reported something already known.

I followed your suggestion: changed the mode to 'background', restarted sublime with debug still set to true.

Once ST starts up, this is what the console shows after loading all plugins (there are no errors):
`...`
`reloading plugin SublimeLinter-contrib-htmlhint.linter`
`reloading plugin SublimeLinter-csslint.linter`
`reloading plugin SublimeLinter-eslint.linter`
`reloading plugin SublimeLinter-json.linter`
`reloading plugin SublimeLinter.__init__`
`reloading plugin SublimeLinter.busy_indicator_view`
`reloading plugin SublimeLinter.commands`
`reloading plugin SublimeLinter.goto_commands`
`reloading plugin SublimeLinter.highlight_view`
`reloading plugin SublimeLinter.log_handler`
`reloading plugin SublimeLinter.message_view`
`reloading plugin SublimeLinter.panel_view`
`reloading plugin SublimeLinter.status_bar_view`
`reloading plugin SublimeLinter.sublime_linter`
`reloading plugin SublimeLinter.tooltips_view`
`reloading plugin SublimeServer.SublimeServer`
`...`
`plugins loaded`
`SublimeLinter: SublimeLinter.sublime-package:71: Logging installed; log level INFO`
`SublimeLinter: SublimeLinter.sublime-package:62: debug mode: on`
`SublimeLinter: SublimeLinter.sublime-package:63: version: 4.0.1`
`reloading settings Packages/User/SublimeServer.sublime-settings`
`Package Control: Skipping automatic upgrade, last run at 2018-03-02 23:14:43, next run at 2018-03-03 00:14:43 or after`

A) opened a .js file: no new events in the console
B) typed something on the opened .js file: no new events in the console.
C) saved the opened .js file: no new events in the console.

**Comment by braver** [comment]
Probably similar issue to this: https://github.com/SublimeLinter/SublimeLinter/issues/1041#issuecomment-370073577

**Comment by kaste** [comment]
I think I took the wrong emoticon. So, I'm sorry for that stupid bug, not you should be sorry for the report. But I think in the meantime a hotfix has been shipped. Otherwise try something like this. Only the selector part is relevant here. 

![image](https://user-images.githubusercontent.com/8558/36925688-4fe5ebe4-1e74-11e8-85f6-727a7415359f.png)

**Comment by josezulu** [comment]
@braver @kaster thanks for your help on this issue.

SublimeLinter updated to 4.0.2 just now, still didn't solve the problem, at least for me.
After the update, I added `"selector": "source.js"` to the linter section: **this solved it!**

**Comment by betorobson** [comment]
select source.js works for me. I was not able to get what is it stand for, could someone tell me?
```json
{
"selector": "source.js"
}
```

**Comment by braver** [comment]
Selectors are how plugins and color schemes figure what kind of code they're looking at. So, javascript files match the "source.js" selector, python files the "source.python", python functions "source.python meta.function", etc etc.

**Comment by betorobson** [comment]
tks :-D
